store/tikv/region_cache.go

```diff
@@ -305,10 +307,26 @@ func (c *RegionCache) asyncCheckAndResolveLoop() {
 		case <-c.notifyCheckCh:
 			needCheckStores = needCheckStores[:0]
 			c.checkAndResolve(needCheckStores)
+		case <-ticker.C:
+			// refresh store once a minute to update labels
+			c.markAllNeedCheck()
 		}
 	}
 }
 
+func (c *RegionCache) markAllNeedCheck() {
+	c.storeMu.Lock()
+	defer c.storeMu.Unlock()
+	if len(c.storeMu.stores) < 1 {
+		return
+	}
+	// we only send notify once
+	for _, s := range c.storeMu.stores {
+		s.markNeedCheck(c.notifyCheckCh, false)
+	}
+	c.notifyCheckCh <- struct{}{}
```

Comment:
Will it cause a deadlock?
1. Before `markAllNeedCheck`, `notifyCheckCh` is empty. 
2. In the middle of this `markAllNeedCheck`, `notifyCheckCh` is filled by another `markAllNeedCheck`.
3. `c.notifyCheckCh <- struct{}{}` waits forever, and `asyncCheckAndResolveLoop` can't consume it.

Reply:
I think there exist the situation that:

1. In the first loop time, execute `markAllNeedCheck` and cost 1min
2. In the second loop time, execute `markAllNeedCheck` again then get stuct in the 
```golang
c.notifyCheckCh <- struct{}{}
```

Reply:
fixed now.

Reply:
No, it won't happen in your case.
"another markAllNeedCheck" in my example, I mean the `markAllNeedCheck` called by `OnSendFail` or `onRegionError`.

Reply:
I think `markAllNeedCheck` won't get sturct as I send the signal like following: @djshow832 

```golang
select {
	case c.notifyCheckCh <- struct{}{}:
	default:
}
```